app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb

```diff
@@ -28,6 +28,10 @@ def destroy
           process_response @hostgroup_class.destroy_all
         end
 
+        def resource_class
+          ForemanPuppet::HostgroupClass
```

Comment:
Why are these needed? :thinking:

Reply:
to return `ForemanPuppet::HostgroupClass` instead of `::HostgroupClass` from core

```ruby
From: /Users/kamilszubrycht/Documents/Development/foreman_puppet/app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb:33 ForemanPuppet::Api::V2::HostgroupClassesController#resource_class:

    31: def resource_class
    32:   binding.pry
 => 33:   ForemanPuppet::HostgroupClass
    34: end

[1] pry(#<ForemanPuppet::Api::V2::HostgroupClassesController>)> super
=> HostgroupClass(id: integer, hostgroup_id: integer, puppetclass_id: integer, hostgroup_puppet_facet_id: integer)
[2] pry(#<ForemanPuppet::Api::V2::HostgroupClassesController>)> ForemanPuppet::HostgroupClass
=> ForemanPuppet::HostgroupClass(id: integer, hostgroup_id: integer, puppetclass_id: integer, hostgroup_puppet_facet_id: integer)
```

Reply:
oh right, in API we have wrong namespace, so it doesn't get guessed correctly.